### ISSUE TITLE
OptimizeResult.fun attribute does not contain the smallest value encountered during optimization using scipy.optimize.minimize() 

### ISSUE THREAD

**Issue by jjkazana** [scipy/scipy#10324]
After comparing values returned in OptimizeResult object with logged function values I found, that it's .fun attribute doesn't contain neither smallest nor biggest encountered (and logged) value for method(s) (success):    
- BFGS (False)
- CG (False) 
- L-BFGS-B (False) 
- Powell (True)
- SLSQP (False) 
- TNC (False)
- Nelder-Mead (False)
   
Failure was expected, as value to be returned was Neural Network's training loss. One may consider my function a black box, I do for sure.

All failed methods ended because of reaching iteration or function evaluation limit according to returned messages. All had "maxiter" option set by hand.

### Function passed to scipy.optimize.minimize():

```
def f(w,data,labels,model, results):
	epochs = len(data)
	rows = len(data[0])
	colls = len(data[0][0])

	wdata = zeros((epochs, rows, rows),  )
	weights = normalised(w, rows, colls)
	
	for i in range(0,len(data)):
		wdata[i] = matmul(data[i],weights.transpose())

	ret = model.fit(wdata, labels, epochs=5)
        
        #logging result to be saved after minimize() returns
	results.append(ret.history['loss'][-1]*100) 

	reset_weights(model)
	return ret.history['loss'][-1]*100
```
###My main function was like:

```
def Run():
    '''setting all variables, models, paths to read files etc.  results = []'''

    res = minimize(f, w, args=(data,labels,model, results))

    '''
    ...
    save res.fun to "res.csv"
    ... 
 
    save results array to "evals.csv"
    save w to file
    '''

And after entire script finished I had in evals.csv values both below and above .fun value from res.csv. So question is what was returned in the res object.

Run() functions run in parallel, however every thread had it's own set of variables and paths and was calling minimize() function on it's own, and all mentioned methods returned some results without crashing. 


### Scipy/Numpy/Python version information:
1.2.1 1.16.2 sys.version_info(major=3, minor=6, micro=7, releaselevel='final', serial=0)

**Comment by andyfaff** [comment]
It would be good if a minimum working example could be provided, so we can fix this issue.

**Comment by mdhaber** [comment]
@andyfaff Shall we close this as a wontfix given our decision about gh-12120?

**Comment by andyfaff** [comment]
Yes. We should have an wiki entry that points to the rationale for behaviour.
